pkg/crdpuller/discovery.go

```diff
@@ -413,6 +413,12 @@ var knownPackages map[string]map[string]apiextensionsv1.JSONSchemaProps = map[st
 		// generate validation for it.
 		"ObjectMeta": apiextensionsv1.JSONSchemaProps{
 			Type: "object",
+
+			// CHANGE vs the OperatorSDK code:
```

Comment:
This solves the issue for Deployments, and anything else in apps/v1, but does it help users' own CRDs? Or other k8s core types that happen to have a `metadata` field?

Reply:
yes, it does. The problem would occur for any metadata field that is not at the top-level, and the fix fixes it in all cases, since it is in the CRD generation itself.

Reply:
Got it, thanks for clarifying. 👍